bundles/org.openhab.binding.homewizard/src/main/java/org/openhab/binding/homewizard/internal/HomeWizardHandler.java

```diff
@@ -0,0 +1,274 @@
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * The {@link HomeWizardHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Daniël van Os - Initial contribution
+ */
+@NonNullByDefault
+public class HomeWizardHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(HomeWizardHandler.class);
+
+    private HomeWizardConfiguration config = new HomeWizardConfiguration();
+
+    private final Lock pollingJobLock = new ReentrantLock();
+    private @Nullable ScheduledFuture<?> pollingJob;
+    private int currentRefreshDelay = 0;
+
+    private final GsonBuilder builder = new GsonBuilder();
+    private final Gson gson = builder.create();
+
+    private final Lock configurationLock = new ReentrantLock(); // to protected fields accessed in init/dispose AND the
+                                                                // poller
+    private String apiURL = "";
+
+    private String meterModel = "";
+    private int meterVersion = 0;
+
+    /**
+     * Constructor
+     *
+     * @param thing The thing to handle
+     */
+    public HomeWizardHandler(Thing thing) {
+        super(thing);
+    }
+
+    /**
+     * Not listening to any commands.
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    /**
+     * If a host has been specified start polling it
+     */
+    @Override
+    public void initialize() {
+        config = getConfigAs(HomeWizardConfiguration.class);
+
+        if (configure()) {
+            stopPolling(true);
+            startPolling();
+        }
+    }
+
+    /**
+     * Handle updates to the configuration gracefully
+     */
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters) {
+        Object cfgObject = configurationParameters.get("refreshDelay");
+        if (cfgObject != null) {
+            BigDecimal bd = (BigDecimal) cfgObject;
+            config.refreshDelay = bd.intValue();
+        }
+
+        cfgObject = configurationParameters.get("ipAddress");
+        if (cfgObject != null) {
+            config.ipAddress = (String) cfgObject;
+        }
+
+        if (configure()) {
+            // If the new configuration is proper, stop the poller if needed
+            stopPolling(true);
+            // Then start it again if it was stopped
+            startPolling();
+        } else {
+            // Stop polling if the new config is invalid
+            stopPolling(false);
+        }
+    }

```

Comment:
Just remove the override of `handleConfigurationUpdate`. When the configuration changes `initialize` is called by the framework. So what do you win by doing all this yourself?

Reply:
My idea was to prevent the reinit (which seemed to work). I would change to config and only stop/start the things that actually needed to be restarted depending on which values changed.

I do agree that it adds complexity to fix something that works quite well without a fix, so I've removed it.